imgui-lwjgl3/src/main/java/imgui/glfw/ImGuiImplGlfw.java

```diff
@@ -0,0 +1,427 @@
+        final ImGuiIO io = ImGui.getIO();
+        if ((io.getConfigFlags() & ImGuiConfigFlags.NavEnableGamepad) == 0) {
+            return;
+        }
+
+        final ByteBuffer buttons = glfwGetJoystickButtons(GLFW_JOYSTICK_1);
+        final int buttonsCount = buttons.limit();
+
+        final FloatBuffer axis = glfwGetJoystickAxes(GLFW_JOYSTICK_1);
+        final int axisCount = axis.limit();
+
+        mapButton(ImGuiNavInput.Activate, 0, buttons, buttonsCount, io);     // Cross / A
+        mapButton(ImGuiNavInput.Cancel, 1, buttons, buttonsCount, io);     // Circle / B
+        mapButton(ImGuiNavInput.Menu, 2, buttons, buttonsCount, io);     // Square / X
+        mapButton(ImGuiNavInput.Input, 3, buttons, buttonsCount, io);     // Triangle / Y
+        mapButton(ImGuiNavInput.DpadLeft, 13, buttons, buttonsCount, io);    // D-Pad Left
+        mapButton(ImGuiNavInput.DpadRight, 11, buttons, buttonsCount, io);    // D-Pad Right
+        mapButton(ImGuiNavInput.DpadUp, 10, buttons, buttonsCount, io);    // D-Pad Up
+        mapButton(ImGuiNavInput.DpadDown, 12, buttons, buttonsCount, io);    // D-Pad Down
+        mapButton(ImGuiNavInput.FocusPrev, 4, buttons, buttonsCount, io);     // L1 / LB
+        mapButton(ImGuiNavInput.FocusNext, 5, buttons, buttonsCount, io);     // R1 / RB
+        mapButton(ImGuiNavInput.TweakSlow, 4, buttons, buttonsCount, io);     // L1 / LB
+        mapButton(ImGuiNavInput.TweakFast, 5, buttons, buttonsCount, io);     // R1 / RB
+        mapAnalog(ImGuiNavInput.LStickLeft, 0, -0.3f, -0.9f, axis, axisCount, io);
+        mapAnalog(ImGuiNavInput.LStickRight, 0, +0.3f, +0.9f, axis, axisCount, io);
+        mapAnalog(ImGuiNavInput.LStickUp, 1, +0.3f, +0.9f, axis, axisCount, io);
+        mapAnalog(ImGuiNavInput.LStickDown, 1, -0.3f, -0.9f, axis, axisCount, io);
+
+        if (axisCount > 0 && buttonsCount > 0) {
+            io.setBackendFlags(io.getBackendFlags() | ImGuiBackendFlags.HasGamepad);
+        } else {
+            io.setBackendFlags(io.getBackendFlags() & ~ImGuiBackendFlags.HasGamepad);
+        }
+    }
+
+    private void mapButton(final int navNo, final int buttonNo, final ByteBuffer buttons, final int buttonsCount, final ImGuiIO io) {
+        if (buttonsCount > buttonNo && buttons.get(buttonNo) == GLFW_PRESS) {
+            io.setNavInputs(navNo, 1.0f);
+        }
+    }
+
+    private void mapAnalog(
+        final int navNo,
+        final int axisNo,
+        final float v0,
+        final float v1,
+        final FloatBuffer axis,
+        final int axisCount,
+        final ImGuiIO io
+    ) {
+        float v = axisCount > axisNo ? axis.get(axisNo) : v0;
+        v = (v - v0) / (v1 - v0);
+        if (v > 1.0f) {
+            v = 1.0f;
+        }
+        if (io.getNavInputs(navNo) < v) {
+            io.setNavInputs(navNo, v);
+        }
+    }
+
+    /**
+     * Updates {@link ImGuiIO} and {@link org.lwjgl.glfw.GLFW} state.
+     */
+    public void newFrame() {
```

Comment:
You should clean up https://github.com/ambye85/imgui-java/blob/master/imgui-lwjgl3/src/test/java/ImGuiGlfwExample.java#L223 method from things you're already doing here. Like setting of display size, updating mouse positioning etc.

Reply:
Completely missed that. Thanks.